### PR TITLE
Misc. cleanups.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/lines.py [hunk 1]
```diff
@@ -657,7 +657,7 @@ def set_data(self, *args):
         *args : (N, 2) array or two 1D arrays
         """
         if len(args) == 1:
-            x, y = args[0]
+            (x, y), = args
```
Comment (timhoffm) [inline lib/matplotlib/lines.py]: Not sure if that's really better in this context.

Comment (anntzer) [reply lib/matplotlib/lines.py]: I prefer this form as it's immediately clear that you're not dropping anything from args, i.e. that single line also asserts that args has length 1.  (Yes, it's redundant with the line just above, but I still think it helps the readability.)

File: lib/matplotlib/backends/backend_pdf.py [hunk 2]
```diff
@@ -1896,7 +1896,7 @@ def draw_tex(self, gc, x, y, s, prop, angle, ismath='TeX!', mtext=None):
         fontsize = prop.get_size_in_points()
         dvifile = texmanager.make_dvi(s, fontsize)
         with dviread.Dvi(dvifile, 72) as dvi:
-            page = next(iter(dvi))
+            page, = dvi
```
Comment (tacaswell) [inline lib/matplotlib/backends/backend_pdf.py]: Are we sure this should only ever contain one page?

Comment (tacaswell) [reply lib/matplotlib/backends/backend_pdf.py]: Yes, because we make it 2 lines above.

**Pull request merged**